File: frontend/src/api/doctors.ts
import client from './client';
import type { Doctor, DoctorBrief } from '../types';

export function fetchDoctors(params?: { department?: string; status?: string }) {
  return client.get<DoctorBrief[]>('/doctors', { params }).then((r) => r.data);
}

export function fetchDoctor(id: number) {
  return client.get<Doctor>(`/doctors/${id}`).then((r) => r.data);
}

export function createDoctor(data: Partial<Doctor>) {
  return client.post<Doctor>('/doctors', data).then((r) => r.data);
}

export function updateDoctor(id: number, data: Partial<Doctor>) {
  return client.put<Doctor>(`/doctors/${id}`, data).then((r) => r.data);
}

export function updateDoctorStatus(id: number, status: string) {
  return client.patch<Doctor>(`/doctors/${id}/status`, { status }).then((r) => r.data);
}

export function deleteDoctor(id: number) {
  return client.delete(`/doctors/${id}`).then((r) => r.data);
}

File: backend/app/routers/doctors.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database import get_db
from app.middleware.auth import require_role
from app.schemas.doctor import DoctorCreate, DoctorUpdate, DoctorStatusUpdate, DoctorResponse, DoctorBrief
from app.services import doctor_service

router = APIRouter(
    prefix="/api/doctors",
    tags=["doctors"],
    dependencies=[Depends(require_role("super_admin"))],
)


@router.get("", response_model=list[DoctorBrief])
def list_doctors(department: str = None, status: str = None, db: Session = Depends(get_db)):
    return doctor_service.list_doctors(db, department=department, status=status)


@router.get("/{doctor_id}", response_model=DoctorResponse)
def get_doctor(doctor_id: int, db: Session = Depends(get_db)):
    doctor = doctor_service.get_doctor(db, doctor_id)
    if not doctor:
        raise HTTPException(status_code=404, detail="医生不存在")
    return doctor


@router.post("", response_model=DoctorResponse, status_code=201)
def create_doctor(data: DoctorCreate, db: Session = Depends(get_db)):
    return doctor_service.create_doctor(db, data)


@router.put("/{doctor_id}", response_model=DoctorResponse)
def update_doctor(doctor_id: int, data: DoctorUpdate, db: Session = Depends(get_db)):
    doctor = doctor_service.update_doctor(db, doctor_id, data)
    if not doctor:
        raise HTTPException(status_code=404, detail="医生不存在")
    return doctor


@router.patch("/{doctor_id}/status", response_model=DoctorResponse)
def update_doctor_status(doctor_id: int, data: DoctorStatusUpdate, db: Session = Depends(get_db)):
    doctor = doctor_service.update_status(db, doctor_id, data.status)
    if not doctor:
        raise HTTPException(status_code=404, detail="医生不存在")
    return doctor


@router.delete("/{doctor_id}")
def delete_doctor(doctor_id: int, db: Session = Depends(get_db)):
    ok = doctor_service.delete_doctor(db, doctor_id)
    if not ok:
        raise HTTPException(status_code=404, detail="医生不存在")
    return {"ok": True}
